src/script_movement.c

```diff
@@ -205,13 +206,34 @@ static void ScriptMovement_MoveObjects(u8 taskId)
     }
 }
 
+// from event_object_movement
+#define sTypeFuncId data[1]
+#define sTimer      data[5]
+
 static void ScriptMovement_TakeStep(u8 taskId, u8 moveScrId, u8 objEventId, const u8 *movementScript)
 {
     u8 nextMoveActionId;
+    struct ObjectEvent *obj = &gObjectEvents[objEventId];
 
-    if (ObjectEventIsHeldMovementActive(&gObjectEvents[objEventId])
-     && !ObjectEventClearHeldMovementIfFinished(&gObjectEvents[objEventId]))
+    if (ObjectEventIsHeldMovementActive(obj) &&
+        !ObjectEventClearHeldMovementIfFinished(obj))
```

Comment:
```suggestion
    if (ObjectEventIsHeldMovementActive(obj) && !ObjectEventClearHeldMovementIfFinished(obj))
```
This is definitely short enough to find on a single line